NUX Signup: Integrate the site topic step into the onboarding flow

southp opened the pull request:
#### Changes proposed in this Pull Request

This PR plugs the site topic picking step into the `onboarding` flow.

#### Parent PR
https://github.com/Automattic/wp-calypso/pull/28455

#### Testing instructions

1. Access http://calypso.localhost:3000/start/onboarding/
1. It should be the 2nd or the 3rd step depending on which portion you belong according to the abtest value of `signupSegmentationStep`.
1. At the survey form, there shouldn't be a vertical field.
1. You should be able to complete the whole signup flow as expected.

southp commented:
@fditrapani Thanks for the review!

> There is a spacing issue above the text field on this PR that @ramonjd is addressing in this PR: #28182

For easier testing, I've applied the same fix in 55c7d6d so it looks like this now:
<img width="824" alt="2018-11-14 3 19 18" src="https://user-images.githubusercontent.com/1842898/48472074-637cd600-e830-11e8-9274-93b79ca56544.png">
I know there will be conflict afterwards, but I'm happy to fix it.

> In that PR, I also mentioned the flow of the copy, mostly the titles, is off from step to step. We agreed to address that in a separate PR once we have this step, and the others, implemented.

Noted. 

> In our latest version of the signup flow, we have decided to remove the "Search above to continue" copy and replace it with a search icon. For now you can remove the copy and similar to above, we can do a separate design pass where we add the search icon to the field.

Removed in dea8404

> The topic I select disappears if I proceed to the next step and then go back again.

Fixed in 7df73ad.

fditrapani commented:
Thanks for those changes @southp. They're looking good. I agree with @michaeldcain about the back button issue.

southp commented:
@michaeldcain By adding `saveSignupStep()` call to `about`, `site-type` and `site-topic` in 8be84a5, it works as expected for me. Could you help me verify if I'm doing it the right way?

fditrapani commented:
@southp when I try this on Calypso live, I'm still getting taken back 2 steps. Here are some things I tried:

1/ Pick a segment
2/ Enter a vertical
3/ Hit back on site title screen
4/ Goes back to segment
5/ Pick segment and see the vertical field populated (that's working :) )

1/ Pick a segment
2/ Enter a vertical
3/ Enter a title
4/ Hit back on the domains
5/ Goes back to vertical screen.

michaeldcain commented:
@southp @fditrapani: the `saveSignupStep()` fix seems to work for me. I tried both the segmenting/and non-segmenting variants of the `/onboarding/` flow with different stages of information filled out. I also reset the branch for calypso.live and it seems to be working correctly now… maybe it was stuck on an old commit for Filippo. I think it's a good fix for now - let's add a card to the backlog to track moving the saving to the flow level.

@southp there's a failing test that needs to be addressed.

southp commented:
I've introduced varying screen copies according to `siteType` in 42dd91a and made up a default.

When there is the site type step:
![demo-site-topic-varying](https://user-images.githubusercontent.com/1842898/48603554-ca75c880-e9b1-11e8-8da5-2b919e6a2eaa.gif)

When there is no and the default shows:
![demo-site-type-exclude](https://user-images.githubusercontent.com/1842898/48603614-fdb85780-e9b1-11e8-9466-4b9f6cca17b3.gif)

Just wanted to build up the foundation first. Hopefully it helps.

michaeldcain commented:
> this uses hints from lib/signup/hint-data.js at the moment, I understand that we want to use the API response to populate this dropdown before we release it to users, right?

@adamziel: that's not a requirement for this card. The endpoint is tracked in a separate card, and we can update it later.

> For this, we will need to exclude site-topic in flows.js:

@niranjan-uma-shankar @fditrapani: I don't think we should exclude the step, just edit the copy as James did. The verticals field already exists for all users - we don't need to test the same functionality.

fditrapani commented:
> I don't think we should exclude the step, just edit the copy as James did. The verticals field already exists for all users - we don't need to test the same functionality.

@michaeldcain the new copy @southp introduced works better for that screen.

michaeldcain commented:
@southp: I spoke to Filippo. The copy change is good. If you can add the `InfoPopover` and center the button from @niranjan-uma-shankar's review, we can go ahead and merge this.

southp commented:
Thanks for all the help, folks :)
I've addressed the feedbacks and it now looks like this:
![demo-updated-site-topic](https://user-images.githubusercontent.com/1842898/48684343-317ec180-ebec-11e8-9f6f-9b9233f94eb2.gif)

michaeldcain commented on client/signup/steps/site-topic/index.jsx:
```diff
@@ -54,36 +63,71 @@ class SiteTopicStep extends Component {
+		const commonSubHeaderText = translate( "Don't stress, you can change this later." );
+
+		switch ( siteType ) {
+			case 'Business':
```
We'll need to update this after #28672 merges.

southp commented:
@michaeldcain Thanks for catching it. I've added it in 79e6550.
